src/ape/api/transactions.py

```diff
@@ -243,3 +274,219 @@ def await_confirmations(self) -> "ReceiptAPI":
+            decoded_value = f'"{value}"'
+
+        elif isinstance(value, (list, tuple)):
+            return [self._decode_value(v) for v in value]
+
+        elif isinstance(value, Struct):
+            return {k: v for k, v in value.items()}
+
+        return decoded_value
+
+
+@dataclass()
+class _MethodTraceSignature:
+    contract_name: str
+    method_name: str
+    arguments: Dict
+    return_value: Any
+
+    def __str__(self) -> str:
+        call_path = f"{self.contract_name}.[{_METHOD_NAME_TRACE_COLOR}]{self.method_name}[/]"
+        arguments_str = self._build_arguments_str()
+        signature = f"{call_path}{arguments_str}"
+
+        return_str = self._build_return_str()
+        if return_str:
+            signature = f"{signature} -> {return_str}"
+
+        return signature
+
+    def _build_arguments_str(self) -> str:
+        if not self.arguments:
+            return "()"
+
+        return _dict_to_str(self.arguments, _ARGUMENT_VALUE_TRACE_COLOR)
+
+    def _build_return_str(self) -> Optional[str]:
+        if self.return_value in [None, [], (), {}]:
+            return None
+
+        elif isinstance(self.return_value, dict):
+            return _dict_to_str(self.return_value, _RETURN_VALUE_TRACE_COLOR)
+
+        elif isinstance(self.return_value, (list, tuple)):
+            return f"[{_RETURN_VALUE_TRACE_COLOR}]{_list_to_str(self.return_value)}[/]"
+
+        return f"[{_RETURN_VALUE_TRACE_COLOR}]{self.return_value}"
+
+
+def _dict_to_str(dictionary: Dict, color: str) -> str:
+    length = sum([len(str(v)) for v in [*dictionary.keys(), *dictionary.values()]])
+    do_wrap = length > _WRAP_THRESHOLD
+
+    index = 0
+    end_index = len(dictionary) - 1
+    arguments_str = "(\n" if do_wrap else "("
+
+    for argument, value in dictionary.items():
+        if do_wrap:
+            arguments_str += _SPACING
+
+        if isinstance(value, (list, tuple)):
+            value = _list_to_str(value, 1 if do_wrap else 0)
+
+        arguments_str += (
+            f"{argument}=[{color}]{value}[/]"
+            if argument and not argument.isnumeric()
+            else f"[{color}]{value}[/]"
+        )
+        if index < end_index:
+            arguments_str += ", "
+
+        if do_wrap:
+            arguments_str += "\n"
+
+        index += 1
+
+    return f"{arguments_str})"
+
+
+def _list_to_str(ls: Union[List, Tuple], depth: int = 0) -> str:
+    if not isinstance(ls, (list, tuple)) or len(str(ls)) < _WRAP_THRESHOLD:
+        return str(ls)
+
+    else:
+        value = "[\n"
+        num_values = len(ls)
+        for index in range(num_values):
+            ls_spacing = _SPACING * (depth + 1)
+            value += f"{ls_spacing}{ls[index]}"
+            if index < num_values - 1:
+                value += ","
+
+            value += "\n"
+
+        value += _SPACING * depth
+        value += "]"
+        return value
+
+
+def _get_method_called(selector: str, contract_type: ContractType) -> Optional[MethodABI]:
```

Comment:
This might be a good util method?

Reply:
here is a better solution for your consideration, you can do `contract_type.mutable_methods[calldata]`
https://github.com/ApeWorX/ethpm-types/pull/34

Reply:
Thanks! This was implemented